MvvmCross-iOSSupport/MvvmCross.iOS.Support.XamarinSidebar/MvxSidebarPresenter.cs

```diff
@@ -33,94 +32,77 @@ protected virtual bool PresentationHintHandler(MvxPanelPresentationHint hint)
                     break;
             }
 
-            if (!viewPresentationAttribute.ShowPanel)
+            if (!attribute.ShowPanel)
             {
-                var menu = Mvx.Resolve<IMvxSideMenu>();
-                menu?.Close();
+                var menu = Mvx.Resolve<IMvxSidebarViewController>();
+                menu?.CloseMenu();
             }
         }
 
-        public override void Close(IMvxViewModel toClose)
+        protected override void ShowRootViewController(UIViewController viewController, MvxRootPresentationAttribute attribute, MvxViewModelRequest request)
         {
-            if (ParentRootViewController.ViewControllers.Count() > 1)
-                ParentRootViewController.PopViewController(true);
-            else if (RootViewController.NavigationController.ViewControllers.Count() > 1)
-                RootViewController.NavigationController.PopViewController(true);
-            else
-                base.Close(toClose);
-        }
+            // check if viewController is a MvxSidebarPanelController
+            if (viewController is IMvxSidebarViewController)
+            {
+                MasterNavigationController = new MvxNavigationController();
 
-        protected MvxPanelPresentationAttribute GetViewPresentationAttribute(IMvxIosView view)
-        {
-            if (view == null)
-                return default(MvxPanelPresentationAttribute);
+                SideBarViewController = viewController as IMvxSidebarViewController;
+                SideBarViewController.SetNavigationController(MasterNavigationController);
 
-            return view.GetType().GetCustomAttributes(typeof(MvxPanelPresentationAttribute), true).FirstOrDefault() as MvxPanelPresentationAttribute;
-        }
+                SetWindowRootViewController(viewController);
 
-        protected virtual void InitRootViewController()
-        {
-            foreach (var view in Window.Subviews)
-                view.RemoveFromSuperview();
+                Mvx.RegisterSingleton<IMvxSidebarViewController>(SideBarViewController);
+
+                CloseMasterNavigationController();
+                CloseModalNavigationController();
+                CloseSplitViewController();
 
-            MasterNavigationController = new UINavigationController();
+                return;
+            }
 
-            OnMasterNavigationControllerCreated();
+            base.ShowRootViewController(viewController, attribute, request);
+        }
 
-            RootViewController = new MvxSidebarPanelController(MasterNavigationController);
-           
-            SetWindowRootViewController(RootViewController);
+        public override void Close(IMvxViewModel toClose)
+        {
+            // if the current root is a SideBarViewController, delegate close responsibility to it
+            if (SideBarViewController != null && SideBarViewController.CloseChildViewModel(toClose))
```

Comment:
Current implementation of `SidebarViewController.CloseChildViewModel` doesn't take into account the `toClose` parameter, maybe we can get rid of that method and just call `base.Close(toClose)` here. Otherwise make `CloseChildViewModel` just return false and mark it as virtual